internal/bootstrap/git/gogit/gogit.go

```diff
@@ -232,3 +244,26 @@ func (g *GoGit) Path() string {
 func isRemoteBranchNotFoundErr(err error, ref string) bool {
 	return strings.Contains(err.Error(), fmt.Sprintf("couldn't find remote ref %q", ref))
 }
+
+func getOpenPgpEntity(key, passphrase string) (*openpgp.Entity, error) {
+	r, err := os.Open(key)
+	if err != nil {
+		return nil, err
+	}
+
+	entityList, err := openpgp.ReadKeyRing(r)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(entityList) == 0 {
+		return nil, fmt.Errorf("no entity formed")
+	}
+
+	err = entityList[0].PrivateKey.Decrypt([]byte(passphrase))
+	if err != nil {
+		return nil, err
+	}
```

Comment:
You are making an assumption here that a key ring only contains a single key. While for some people this might be true, others may contain multiple, and you probably want to provide people the option to select what key from the ring should be used (using e.g. a fingerprint selector of the public key).